src/cli/adapter/prettier.ts

```diff
@@ -1,12 +1,56 @@
 import path from 'path';
 
 import fs from 'fs-extra';
-import { Options, check, format, getFileInfo, resolveConfig } from 'prettier';
+import {
+  Options,
+  SupportLanguage,
+  check,
+  format,
+  getSupportInfo,
+  resolveConfig,
+} from 'prettier';
 
 import { crawlDirectory } from '../../utils/dir';
 import { Logger, pluralise } from '../../utils/logging';
 import { getConsumerManifest } from '../../utils/manifest';
 
+let languages: SupportLanguage[] | undefined;
```

Comment:
question: is there a reason we are using a global variable?

Reply:
This memoises the `languages` array rather than recalculating it on every file that passes through `inferParser`. We don't vary the plugins between files so the array is always the same.